docs/concepts/unions.md

```diff
@@ -76,25 +76,59 @@ print(User(id='456'))  # (2)
 2. We're in lax mode and the numeric string `'123'` is valid as input to the first member of the union, `int`.
    Since that is tried first, we get the surprising result of `id` being an `int` instead of a `str`.
 
+!!! tip
+    For complex cases, it's recommended to use either [`left_to_right`](#left-to-right-mode) mode or [discriminated unions](#discriminated-unions).
+    With these approaches, you can control which member of the union to validate against, and in which order.
+    If you're looking for incredibly specialized behavior, you can use a [custom validator](../concepts/validators.md#field-validators).
+
 ### Smart Mode
 
 Because of the surprising side effects of `union_mode='left_to_right'`, in Pydantic >=2 the default mode for `Union` validation is `union_mode='smart'`.
```

Comment:
IMO it's not "side effects" (I know not from this PR but I've wanted this line changed for a while :) )

```suggestion
Because of the potentially surprising results of `union_mode='left_to_right'`, in Pydantic >=2 the default mode for `Union` validation is `union_mode='smart'`.
```